### PR TITLE
[sc-19213] Add IdentityNow template

### PR DESCRIPTION
**Pull request by richardxgao**
Add the IdentityNow SoR template with the following entity support:
- Account
- Entitlement
- AccountEntitlement
- Identity
- Entitlement -[MEMBER] -> Account

Attributes in this template are not comprehensive; only the ones listed in the [IDN API Docs](https://developer.sailpoint.com/idn/api/v3/list-accounts) are included. Additional custom attributes configured by the user need to be added manually.

### CODE REVIEW

File: identitynow.sgnl.yaml [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+#       "uniqueIDAttribute": "id",
+#       "apiVersion": "beta"
+#     },
+#     "identities": {
+#       "uniqueIDAttribute": "id",
+#       "apiVersion": "beta"
+#     }
+#   }
+# }
+adapterConfig: "ewogICJhcGlWZXJzaW9uIjogInYzIiwKICAiZW50aXR5Q29uZmlnIjogewogICAgImFjY291bnRzIjogewogICAgICAidW5pcXVlSURBdHRyaWJ1dGUiOiAiaWQiCiAgICB9LAogICAgImVudGl0bGVtZW50cyI6IHsKICAgICAgInVuaXF1ZUlEQXR0cmlidXRlIjogImlkIiwKICAgICAgImFwaVZlcnNpb24iOiAiYmV0YSIKICAgIH0sCiAgICAiYWNjb3VudEVudGl0bGVtZW50cyI6IHsKICAgICAgInVuaXF1ZUlEQXR0cmlidXRlIjogImlkIiwKICAgICAgImFwaVZlcnNpb24iOiAiYmV0YSIKICAgIH0sCiAgICAiaWRlbnRpdGllcyI6IHsKICAgICAgInVuaXF1ZUlEQXR0cmlidXRlIjogImlkIiwKICAgICAgImFwaVZlcnNpb24iOiAiYmV0YSIKICAgIH0KICB9Cn0K"
```
Comment (shash-sgnl) [inline identitynow.sgnl.yaml:33]: Add the decoded json here in a single line as a comment.

File: identitynow.sgnl.yaml [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+entities:
+  Account:
+    displayName: Account
+    externalId: accounts
+    description: Account entity in IdentityNow
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 250
```
Comment (shash-sgnl) [inline identitynow.sgnl.yaml:50]: why 250?

**Review by shash-sgnl — CHANGES_REQUESTED** [review]

File: identitynow.sgnl.yaml (hunk 2 above)
Comment (marc-sgnl) [reply identitynow.sgnl.yaml:50]: default and max page size of Sailpoint IDN - https://developer.sailpoint.com/idn/api/v3/list-accounts

File: identitynow.sgnl.yaml (hunk 1 above)
Comment (marc-sgnl) [reply identitynow.sgnl.yaml:33]: Curious -- is there a reason to add it as a single line? This is one of the more complex adapter configs (even moreso if it has filters) lines 13 to 32 articulate that decoded json pretty neatly. Wouldn't it make for a worse developer experience to have it pulled into one line?

File: identitynow.sgnl.yaml [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        externalId: identityStatus
+        type: String
+      - name: isManager
+        externalId: isManager
+        type: Bool
+      - name: lastRefresh
+        externalId: lastRefresh
+        type: DateTime
+
+relationships:
```
Comment (marc-sgnl) [inline identitynow.sgnl.yaml:220]: Suggest creating the relationship between Accounts and an Identity too, via the identityId attribute on accounts to the id attribute on identities.

File: identitynow.sgnl.yaml [hunk 4]
```diff
@@ -0,0 +1,239 @@
+# IdentityNow Configuration YAML for DS2.0
+displayName: "IdentityNow"
```
Comment (marc-sgnl) [inline identitynow.sgnl.yaml:2]:
```suggestion
displayName: "Sailpoint IdentityNow"
```

File: identitynow.sgnl.yaml [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+icon: "PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTQwLjQxMTggNDIuMTM3M0gxMEwzNS4yMzUzIDEwTDQwLjQxMTggNDIuMTM3M1oiIGZpbGw9IiMwMDMzQTEiLz4KPHBhdGggZD0iTTQwLjQxMTggNDIuMTM3M0g1NEwzNS4yMzUzIDEwTDQwLjQxMTggNDIuMTM3M1oiIGZpbGw9IiNDQzI3QjAiLz4KPHBhdGggZD0iTTQwLjQxMTggNDIuMTM3M0gxMEw0Mi4zNTI5IDUzLjc4NDNMNDAuNDExOCA0Mi4xMzczWiIgZmlsbD0iIzAwNzFDRSIvPgo8cGF0aCBkPSJNNDAuNDExOCA0Mi4xMzczSDU0TDQyLjM1MjkgNTMuNzg0M0w0MC40MTE4IDQyLjEzNzNaIiBmaWxsPSIjRTE3RkQyIi8+Cjwvc3ZnPgo="
+description: "IdentityNow as a System of Record"
```
Comment (marc-sgnl) [inline identitynow.sgnl.yaml:4]:
```suggestion
description: "Sailpoint IdentityNow as a System of Record"
```

File: identitynow.sgnl.yaml (hunk 1 above)
Comment (shash-sgnl) [reply identitynow.sgnl.yaml:33]: We have one-liner jsons in other templates + this bloats up the yaml.
But, don't feel strongly about it. This looks good.

Comment (richardxgao) [reply identitynow.sgnl.yaml:33]: Will leave as-is then.

File: identitynow.sgnl.yaml (hunk 3 above)
Comment (richardxgao) [reply identitynow.sgnl.yaml:220]: Added the `AccountIdentity` relationship between `Account -> Identity`. Open to other names.

File: identitynow.sgnl.yaml (hunk 4 above)
Comment (richardxgao) [reply identitynow.sgnl.yaml:2]: Done.

**Review by marc-sgnl — APPROVED** [review]

**Review by shash-sgnl — APPROVED** [review]

**Pull request merged**